lib/ansible/plugins/cliconf/eos.py

```diff
@@ -247,10 +247,15 @@ def get_device_info(self):
 
         device_info['network_os_hostname'] = data['hostname']
 
-        reply = self.get('bash timeout 5 cat /mnt/flash/boot-config')
-        match = re.search(r'SWI=(.+)$', reply, re.M)
-        if match:
-            device_info['network_os_image'] = match.group(1)
+        try:
+            reply = self.get('bash timeout 5 cat /mnt/flash/boot-config')
+
+            match = re.search(r'SWI=(.+)$', reply, re.M)
+            if match:
+                device_info['network_os_image'] = match.group(1)
+        except AnsibleConnectionFailure:
+            # This requires enable mode to run
```

Comment:
Can we emit a warning/info instead of silently failing

Reply:
I think a warning might be a bit strong for this, but I don't know what the implications of this data not being populated are.